portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1947,6 +1947,10 @@ else if (method.equals(Http.Method.HEAD)) {
 				response.addHeader(header.getName(), header.getValue());
 			}
 
+			if (httpEntity == null) {
```

Comment:
Shouldn't you be shortcutting right after here https://github.com/adolfopa/liferay-portal/blob/32ff8f91c82f1239362881c2c2e03eaab3efb036/portal-impl/src/com/liferay/portal/util/HttpImpl.java#L1889 ? Why wait for this much late?

Reply:
Hey @shuyangzhou 

The code is mutating the response object (e.g. status code), so I waited until the end in case there's some code that depend on those side effects.